### PR TITLE
Fix Hatch Color

### LINKED ISSUE #8865: Figure patch does not support hatching
### Bug report

**Bug summary**

Figure patch does not support hatching.

**Code for reproduction**

```python
from pylab import *
from matplotlib.patches import *

fig = gcf()
r = Rectangle((0, 0), 100, 100, )
r.set_hatch("/")
fig.patches.append(r)
fig.patch.set_hatch("/")

show()
```

**Actual outcome**

Only the "r" rectangle (blue, lower left corner) is hatched.

**Expected outcome**

The whole figure also gets hatched.

**Matplotlib version**
  * Matplotlib Version: 2.0.2 and master

Obviously not a big deal but something seems wrong here; Figure.draw has the snippet
```
        try:
            renderer.open_group('figure')
            if self.get_tight_layout() and self.axes:
                try:
                    self.tight_layout(renderer, **self._tight_parameters)
                except ValueError:
                    pass
                    # ValueError can occur when resizing a window.

            if self.frameon:
                self.patch.draw(renderer)  # <--- this

            mimage._draw_list_compositing_images(
                renderer, self, artists, self.suppressComposite)

            renderer.close_group('figure')
        finally:
            self.stale = False
```
so the patch should just get drawn like a normal rectangle (frameon is True).  Note that calling e.g. `set_facecolor` on the figure patch works as expected.

### PR DESCRIPTION
**Pull request by Impaler343**
## PR summary

Closes #8865
Simple fix by checking if face-color is the same as edge-color for a patch, If yes, do not update hatch-color 

Gives the needed output:

image

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by Impaler343** [comment]
I'm not sure about how to solve these code-cov failing tests, can someone help me out?

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -80,8 +80,8 @@ def __init__(self, *,
                     "the edgecolor or facecolor properties.")
             self.set_color(color)
         else:
-            self.set_edgecolor(edgecolor)
             self.set_facecolor(facecolor)
+            self.set_edgecolor(edgecolor)
```
[the remark this answers is not in the document]
Comment (Impaler343) [reply lib/matplotlib/patches.py:84]: Yes, the ```facecolor``` attribute must be defined first to use it later in the ```_set_edgecolor``` method

**Review by davidlowryduda — APPROVED** [review]

**Comment by Impaler343** [comment]
Could somebody take a look at this patch?

**Comment by rcomer** [comment]
Please add a test to verify the new behaviour.  Otherwise future changes might inadvertently break it again.
https://matplotlib.org/devdocs/devel/testing.html

**Comment by timhoffm** [comment]
Updating the hatch color depending on the face color does not feel right. While it may result in hatching for the given example, it may result in strange order dependent behavior. Say, your patch face Color is white and you want to change that to grey with white hatching. With this PR, it only works for

```
patch.set_facecolor(„grey“)
patch.set_edgecolor(„white“)
```
But not if you change the order of both lines.

I’m on mobile, so cannot dig fully into the context, but I have the impression that the underlying problem of the original issue may be that hatch color and face color are the same so that the hatch is drawn but not visible. Can you confirm this? If that is the case, other fixes (or maybe no fix at all) are needed - which depends on the context.

**Comment by Impaler343** [comment]
> Updating the hatch color depending on the face color does not feel right. While it may result in hatching for the given example, it may result in strange order dependent behavior. Say, your patch face Color is white and you want to change that to grey with white hatching. With this PR, it only works for
> 
> ```
> patch.set_facecolor(„grey“)
> patch.set_edgecolor(„white“)
> ```
> 
> But not if you change the order of both lines.
> 
> I’m on mobile, so cannot dig fully into the context, but I have the impression that the underlying problem of the original issue may be that hatch color and face color are the same so that the hatch is drawn but not visible. Can you confirm this? If that is the case, other fixes (or maybe no fix at all) are needed - which depends on the context.

Yes you are right, the hatch color is set based on the edge color which is set to (0,0,0,0). Hence, the patch. I probably need to default to rcParams if the face color and edge colors match?

**Comment by timhoffm** [comment]
Ok, so the issue here is that the default edge color for the figure patch  is such that the hatching is invisible. I'd then claim that everything works as intended and we should close both this PR and the issue. The resolution to the issue would be on the user side to set an edge color when they want hatching. Ping @anntzer for confirmation.

**Comment by anntzer** [comment]
Yes, looking at it again the issue was just invalid and arising because the patch edgecolor also defaults to white.  Thanks for resurrecting the discussion, though!

**Comment by Impaler343** [comment]
> The resolution to the issue would be on the user side to set an edge color when they want hatching
Should we add this to the docs then? Recommending to always set edge color when setting hatches, as it is confusing to not see a hatch when it is specified explicitly

**Comment by timhoffm** [comment]
This applies only to the figure patch, because that has different defaults. We don’t have specific documentation for the figure patch and IMHO it’s really rare to hatch the figure. I wouldn’t clutter the regular docs with that.

**Pull request closed without merging**